code/CMSProfileController.php

```diff
@@ -67,11 +67,13 @@ public function canView($member = null)
             return false;
         }
 
-        // Check they can access the CMS and that they are trying to edit themselves
-        if (Permission::checkMember($member, "CMS_ACCESS")
-            && $member->ID === Security::getCurrentUser()->ID
-        ) {
-            return true;
+        // Check they are trying to edit themselves
+        if ($member->ID === Security::getCurrentUser()->ID) {
+            // Check they can access via self::$required_permission_codes
+            $codes = $this->getRequiredPermissions();
+
+            // allow explicit TRUE || default to Permission::checkMember()
+            return $codes === true || Permission::checkMember($member, $codes);
```

Comment:
$codes should be an array - but this logic is actually defined in the parent function, so I think this whole thing can be changed to:

`return $member->ID === Security::getCurrentUser()->ID && parent::canView($member);`